cmd/virt-launcher/virt-launcher.go

```diff
@@ -493,6 +509,10 @@ func main() {
 			finalShutdownCallback,
 			gracefulShutdownCallback)
 
+		if err := runDownwardMetricsVirtioServer(signalStopChan); err != nil {
```

Comment:
I'd still move that up to earlier in the process.

Reply:
Moved some lines above, but this is as above as we can go, unfortunately.

Reply:
As stated here https://github.com/kubevirt/kubevirt/pull/14067#discussion_r2063088629, the downwardMetrics server startup expects the file `launcher-sock` to be present, otherwise the startup will fail. `markReady` function creates the `launcher-sock` file see:

https://github.com/kubevirt/kubevirt/blob/main/cmd/virt-launcher/virt-launcher.go#L73

Therefore, if we try to spin the downwardMetrics server before `markReady` function is called, the downwardMetrics server startup will always fail.

We can drop this check in the server startup and leave the runtime check, the server use this socket to contact libvirt to fetch the metrics, and it fails if the socket is not available (it will retry in the next request). However, I'm not sure if this would make sense. WDYT?

Reply:
Can't it handle the sock not existing, or did I misunderstand this?

https://github.com/kubevirt/kubevirt/pull/14067#discussion_r2058172372

Reply:
Okay, I see. These are two different sockets. Right after `markReady` should be the best place to start it then.